From 6cecc20831eacec397a8c88b76757aeb662621e2 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Thu, 17 May 2012 01:18:57 -0400
Subject: [PATCH] osd_login: fix DESTDIR install
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

If the destination dir doesn't exist, the current install fails:
make  install-data-hook
make[4]: Entering directory `/var/tmp/portage/net-fs/nfs-utils-1.2.6/work/nfs-utils-1.2.6/utils/osd_login'
/usr/bin/install -c --mode 755 osd_login /var/tmp/portage/net-fs/nfs-utils-1.2.6/image//sbin/osd_login
/usr/bin/install: cannot create regular file ‘/var/tmp/portage/net-fs/nfs-utils-1.2.6/image//sbin/osd_login’: No such file or directory
make[4]: *** [install-data-hook] Error 1

Rewrite the code to use automake's existing script support rather
than trying to install the file by hand.

Reported-by: Branko BAdrljica <brankob@avtomatika.com>
Signed-off-by: Mike Frysinger <vapier@gentoo.org>
---
 utils/osd_login/Makefile.am |   11 +++++------
 1 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/utils/osd_login/Makefile.am b/utils/osd_login/Makefile.am
index adc493a..c85584c 100644
--- a/utils/osd_login/Makefile.am
+++ b/utils/osd_login/Makefile.am
@@ -1,12 +1,11 @@
 ## Process this file with automake to produce Makefile.in
 
-OSD_LOGIN_FILES= osd_login
-
-EXTRA_DIST= $(OSD_LOGIN_FILES)
+# These scripts go in /sbin (not /usr/sbin), and that cannot be
+# overriden at config time.
+sbindir = /sbin
 
-all-local: $(OSD_LOGIN_FILES)
+OSD_LOGIN_FILES= osd_login
 
-install-data-hook:
-	$(INSTALL) --mode 755 osd_login $(DESTDIR)/sbin/osd_login
+sbin_SCRIPTS= osd_login
 
 MAINTAINERCLEANFILES = Makefile.in
-- 
1.7.8.6

